packages/react-router/modules/matchPath.js

```diff
@@ -30,7 +30,8 @@ const matchPath = (pathname, options = {}) => {
   if (typeof options === 'string')
     options = { path: options }
 
-  const { path, exact = false, strict = false } = options
+  const { exact = false, strict = false } = options
+  const path = options.path || options.from
```

Comment:
Nothing. I guess we could create a new component for the use case of `<Redirect>` inside a `<Switch>`, but we could also just tell people "don't do that".